addons/binding/org.openhab.binding.dsmr/src/main/java/org/openhab/binding/dsmr/handler/DSMRMeterHandler.java

```diff
@@ -0,0 +1,206 @@
+     */
+    private DSMRMeter meter;
+
+    /**
+     * Timestamp when last values were received
+     */
+    private long lastValuesReceivedTs;
+
+    /**
+     * Watchdog for checking if the meter is alive
+     */
+    private ScheduledFuture<?> meterWatchdog;
+
+    /**
+     * Creates a new MeterHandler for the given Thing
+     *
+     * @param thing {@link Thing} to create the MeterHandler for
+     */
+    public DSMRMeterHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * DSMR Meter don't support handling commands
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No comments can be handled
+    }
+
+    /**
+     * Initializes a DSMR Meter
+     *
+     * This method will load the corresponding configuration
+     */
+    @Override
+    public void initialize() {
+        logger.debug("Initialize MeterHandler for Thing {}", getThing().getUID());
+
+        Configuration config = getThing().getConfiguration();
+        DSMRMeterConfiguration meterConfig = null;
+
+        // Since the config instance cannot be null we don't check it
+        meterConfig = config.as(DSMRMeterConfiguration.class);
+
+        /*
+         * The meterConfig can only be null if the specified class doesn't have a constructor have 0 arguments.
+         * This is a programming bug. Don't handle this situation
+         */
+        DSMRMeterType meterType = null;
+
+        try {
+            meterType = DSMRMeterType.valueOf(getThing().getThingTypeUID().getId().toUpperCase());
+        } catch (IllegalArgumentException iaEx) {
+            logger.error("Unknown meterType", iaEx);
+
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "This could not be initialized. Delete Thing if the problem persists.");
+
+            return;
+        } catch (NullPointerException npe) {
+            logger.error("Invalid Thing", npe);
+
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "This could not be initialized. Delete Thing if the problem persists.");
+
+            return;
+        }
+
+        meter = new DSMRMeter(new DSMRMeterDescriptor(meterType, meterConfig.channel), this);
+
+        // Initialize meter watchdog
+        meterWatchdog = scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                if (System.currentTimeMillis()
+                        - lastValuesReceivedTs > DSMRMeterConstants.METER_VALUES_RECEIVED_TIMEOUT) {
+                    if (getThing().getStatus() != ThingStatus.OFFLINE) {
+                        updateStatus(ThingStatus.OFFLINE);
+                    }
+                }
+            }
+        }, DSMRMeterConstants.METER_VALUES_RECEIVED_TIMEOUT, DSMRMeterConstants.METER_VALUES_RECEIVED_TIMEOUT,
+                TimeUnit.MILLISECONDS);
+
+        updateStatus(ThingStatus.OFFLINE);
```

Comment:
Why OFFLINE? You should only set it to offline, if it is for sure unreachable. Otherwise set it to UNKNOWN.